control-plane/csi-driver/src/bin/node/nodeplugin_grpc.rs

```diff
@@ -54,6 +63,67 @@ impl NodePlugin for NodePluginSvc {
             device_path: device.devname(),
         }))
     }
+
+    async fn cleanup(
+        &self,
+        request: Request<CleanupRequest>,
+    ) -> Result<Response<CleanupReply>, Status> {
+        let volume_id = request.into_inner().volume_id;
+        debug!("Starting cleanup for volume: {volume_id}");
+
+        let nqn = format!("{}:{volume_id}", nvme_target_nqn_prefix());
+        match Subsystem::try_from_nqn(&nqn) {
+            Ok(subsystem_paths) => {
+                for subsystem_path in subsystem_paths {
+                    debug!("Processing subsystem: {subsystem_path:?}");
+                    if !subsystem_path.state.contains("deleting") {
+                        runtime::spawn_blocking(move || {
+                            subsystem_path
+                                .disconnect()
+                                .map_err(|error| Status::aborted(error.to_string()))
+                        })
+                        .await
+                        .map_err(|error| Status::aborted(error.to_string()))??;
+                    }
+                }
+                Err(Status::internal(format!(
+                    "Cleanup initiated for all stale entries of: {volume_id}. Returning error for validation on retry."
```

Comment:
Why do we return error here?

Reply:
To utilise the retry without timing out, but would refactor this.